src/input.c

```diff
@@ -624,6 +718,82 @@ static void parse_char(input_t *restrict input, char ch)
 
             input->state = STATE_GROUND;
         }
+        break;
+    case STATE_CSI_U:
+        if (ch == '?' || ch == '>' || ch == '<') {
+            input->parm_prefix = ch;
+        } else if (ch >= '0' && ch <= '9') {
+            input->parm = input->parm * 10 + (ch - '0');
+        } else if (ch == ';') {
+            if (input->parm_count < MAX_PARMS)
+                input->parms[input->parm_count++] = input->parm;
+            input->parm = 0;
+        } else if (ch == ':') {
+            /* Sub-parameter delimiter (Kitty Keyboard Protocol) */
+            if (input->parm_count < MAX_PARMS) {
+                int idx = input->parm_count;
+                if (input->sub_parm_count[idx] < 4) {
+                    input->sub_parms[idx][input->sub_parm_count[idx]++] =
+                        input->parm;
+                }
+            }
+            input->parm = 0;
+        } else {
+            if (input->parm_count < MAX_PARMS)
+                input->parms[input->parm_count++] = input->parm;
+
+            if (ch == 'c' && input->parm_prefix == '?') {
+                /* Device attributes */
+                if (input->device_attributes)
+                    free(input->device_attributes);
+                input->device_attributes =
+                    malloc(input->parm_count * sizeof(int));
+                if (input->device_attributes) {
+                    memcpy(input->device_attributes, input->parms,
+                           input->parm_count * sizeof(int));
+                    input->da_count = input->parm_count;
+                }
+                device_attributes_report(input);
+
+            } else if (ch == 'u') {
+                input->csi_u_params.keycode = input->parms[0];
+                /* use press as default event_type */
+                input->csi_u_params.event_type = 1;
+                if (input->parm_count == 2) {
+                    if (input->sub_parm_count[1] > 0) {
+                        input->csi_u_params.event_type = input->parms[1];
+                        input->csi_u_params.modifiers = input->sub_parms[1][0];
+                    } else {
+                        input->csi_u_params.modifiers = input->parms[1];
+                    }
```

Comment:
Problem: According to Kitty Protocol spec, the format is:
```
  CSI keycode ; modifiers:event_type u
```

The code interprets parms[1] as `event_type` when sub-params exist, but `parms[1]` contains the full second parameter (modifiers field), not just event_type. The event_type is in the sub-parameter after the colon, not in `parms[1]`.

Correct interpretation:
- `parms[1] = modifiers` (including offset of 1)
- `sub_parms[1][0] = event_type` (when present after colon)

The logic is inverted.